python/packages/streamsx/rest_primitives.py

```diff
@@ -49,6 +50,21 @@ def __init__(self, json_rep, rest_client):
 
     def __str__(self):
         return pformat(self.__dict__)
+    __repr__ = __str__
+
+    def refresh(self):
+        """Make a REST call to refresh the JSON object stored within the object
+
+        Returns:
+            _ResourceElement: The provided object instance is returned with updated value
+        """
+        rest_client = self.rest_client
+        rest_self = self.rest_self
+        json_rep = rest_client.make_request(rest_self)
+        self.__dict__.clear()
```

Comment:
The `__init__` method currently adds attributes from the returned JSON object directly onto the `_ResourceElement` object.  The design makes it difficult to clear these attributes on subsequent `refresh()`, because I don't keep track of the keys that are added to the object.

Another solution is to make `_ResourceElement` object inherit from `collections.Mapping`, so it acts like a dict object.  This way, we could store the returned JSON in a property, which can be replaced much easily.

The proposed change would alter the way that the JSON fields are accessed.  For example, in the current code, accessing the health of a job would be:
```
job.health
```
In the new proposed solution, it would be:
```
job['health']
#or
job.get('health')
```